[feature][doc] Documentation for Pulsar shell (#16725)
apache/pulsar: site2/docs/reference-cli-tools.md

|`-i`, `--interval`| Interval to collect load report, in milliseconds ||
|`-h`, `--help`| Display help information ||

## `pulsar-shell`

[Pulsar shell](administration-pulsar-shell.md) tool.

### Interactive mode

Usage

```bash

$ pulsar-shell

```

Options

| Flag               | Description                                                               | Default          |
|--------------------|---------------------------------------------------------------------------|------------------|

> | `-c`, `--config`   | Client configuration file. It will be used as `default` config.           | conf/client.conf | 

Reviewer: ```suggestion
| `-c`, `--config`   | Client configuration file. It is used as a `default` config.           | conf/client.conf | 
```

> | `--fail-on-error` | If true, the shell will be interrupted if a command throws an exception.  | false            | 

Reviewer: ```suggestion
| `--fail-on-error` | If true, the shell is interrupted when a command throws an exception.  | false            | 
```

> | `-h`, `--help`     | Show this help                                                            | |

Reviewer: ```suggestion
| `-h`, `--help`     | Show this help.                                                            | |
```

### Non interactive mode

Usage

```bash

$ pulsar-shell -f [FILE]
$ pulsar-shell -e [COMMAND]
$ echo "[COMMAND]" | pulsar-shell -

```

Options

| Flag                      | Description                                                                                         | Default         |
|---------------------------|-----------------------------------------------------------------------------------------------------|-----------------|

> | `-c`, `--config`          | Client configuration file. It will be used as `default` config.                                     | conf/client.conf | 

Reviewer: ```suggestion
| `-c`, `--config`          | Client configuration file. It is used as a `default` config.                                     | conf/client.conf | 
```

> | `--fail-on-error`         | If true, the shell will be interrupted if a command throws an exception.                            | false           | 

Reviewer: ```suggestion
| `--fail-on-error`         | If true, the shell is interrupted when a command throws an exception.                            | false           | 
```

| `-np`, `--no-progress`    | Display raw output of the commands without the fancy progress visualization.                        | false           | 
| `-f`, `--filename`        | Input filename with a list of commands to be executed. Each command must be separated by a newline. |                 |
| `-e`, `--execute-command` | Execute this command and exit.                                                                      | |
| `-` | Read commands from the standard input.                                                              | |

> | `-h`, `--help`     | Show this help                                                                                      | |

Reviewer: ```suggestion
| `-h`, `--help`     | Show this help.                                                                                      | |
```

Commands
* `admin` - See [Admin API](admin-api-overview.md)
* `client` - See [pulsar-client](#pulsar-client)

> * `config`

Reviewer: Any referenced link for this item?

Author: it's in the next line

### `config`

Manage shell configurations.

#### `use`

Use a specific configuration for next commands.

```
default(localhost)> config use mycluster
```

#### `create`

Create a new configuration.

```
default(localhost)> config create --file ./conf/client.conf mycluster
```

Options

| Flag     | Description              | Default         |
|----------|--------------------------|-----------------|
| `--file` | File path of the config. |  | 
| `--url`  | URL of the config.       |  |

> | `--value`  | Inline value of the config. Base64 encoded value is supported with prefix `base64:` |  |

Reviewer: ```suggestion
| `--value`  | Inline value of the config. Base64-encoded value is supported with the prefix `base64:`. |  |
```

#### `update`

Update an existing configuration.

```
default(localhost)> config update --file ./conf/client.conf mycluster
```

Options

| Flag     | Description              | Default         |
|----------|--------------------------|-----------------|
| `--file` | File path of the config. |  | 
| `--url`  | URL of the config.       |  |

> | `--value`  | Inline value of the config. Base64 encoded value is supported with prefix `base64:` |  |

Reviewer: ```suggestion
| `--value`  | Inline value of the config. Base64-encoded value is supported with the prefix `base64:`. |  |
```
